src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -367,7 +370,19 @@ private function addPaginationParameters(bool $v3, ResourceMetadata $resourceMet
                     'required' => false,
                     'description' => 'The number of items per page',
                 ];
-                $v3 ? $itemPerPageParameter['schema'] = ['type' => 'integer'] : $itemPerPageParameter['type'] = 'integer';
+                if ($v3) {
+                    $itemPerPageParameter['schema'] = [
+                        'type' => 'integer',
+                        'default' => $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_items_per_page', 30, true),
+                        'minimum' => 0,
+                    ];
+
+                    if ($resourceMetadata->getCollectionOperationAttribute($operationName, 'maximum_items_per_page', false, true)) {
```

Comment:
you can store the result of `getCollectionOperationAttribute()` in a variable to prevent a useless function call.

Reply:
fixed